components/nodemanager-service/pgdb/nodes.go

```diff
@@ -31,6 +31,13 @@ const (
 	NodeManagersTableAbbrev = "nm"
 )
 
+const sqlGetNodeFields = `
+SELECT DISTINCT %s 
+from nodes n
+%s %s
```

Comment:
i'm reusing the filtering functions we already have in place for GetNodes so followed the same pattern here. 
its basically select distinct {field} from nodes n {where blabla} {having blabla} {and/where field is not empty string}